src/components/ProfilePage/SocialLinks.jsx

```diff
@@ -43,24 +43,36 @@ class SocialLinks extends React.Component {
   }
 
   handleChange(e) {
-    const {
-      name,
-      value,
-    } = e.target;
-
-    if (name !== 'visibility') {
-      const updatedList = this.props.committedValue.map((socialLink) => {
-        if (socialLink.platform === name) {
-          return { platform: name, social_link: value };
-        }
-        return socialLink;
-      });
-      this.props.changeHandler(this.props.formId, 'socialLinks', updatedList);
+    const { name, value } = e.target;
+
+    if (name !== 'visibilitySocialLinks') {
```

Comment:
The social links are a bit special.  If we're updating them, we need to merge them with any existing social link drafts, essentially sending a fresh copy of the whole data structure back to the reducer.  This helps the reducer stay simple and keeps special cases out of it, concentrating them here, where they began.